arcgis-ios-sdk-samples/Scenes/Orbit camera around object/OrbitCameraSettingsViewController.swift

```diff
@@ -0,0 +1,120 @@
+        graphic: AGSGraphic
+    ) {
+        orbitGeoElementCameraController = cameraController
+        planeGraphic = graphic
+        super.init(coder: coder)
+    }
+    
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: Actions
+    
+    func updateUIForHeading() {
+        headingSlider.value = Float(orbitGeoElementCameraController.cameraHeadingOffset)
+        let measurement = Measurement(value: orbitGeoElementCameraController.cameraHeadingOffset, unit: UnitAngle.degrees)
+        headingLabel.text = measurementFormatter.string(from: measurement)
+    }
+    
+    func updateUIForPitch() {
+        let pitch = planeGraphic.attributes["PITCH"] as! Float
+        pitchSlider.value = pitch
+        let measurement = Measurement(value: Double(pitch), unit: UnitAngle.degrees)
+        pitchLabel.text = measurementFormatter.string(from: measurement)
+    }
+    
+    @IBAction func headingValueChanged(_ sender: UISlider) {
+        orbitGeoElementCameraController.cameraHeadingOffset = Double(sender.value)
+        updateUIForHeading()
+    }
+    
+    @IBAction func pitchValueChanged(_ sender: UISlider) {
+        planeGraphic.attributes["PITCH"] = sender.value
+        updateUIForPitch()
+    }
+    
+    @IBAction func distanceInteractionSwitchValueChanged(_ sender: UISwitch) {
+        orbitGeoElementCameraController.isCameraDistanceInteractive = sender.isOn
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        tableViewContentSizeObservation = tableView.observe(\.contentSize) { [unowned self] tableView, _ in
+            self.preferredContentSize = CGSize(width: self.preferredContentSize.width, height: tableView.contentSize.height)
+        }
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        tableViewContentSizeObservation = nil
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Apply initial values to controls.
+        updateUIForHeading()
+        updateUIForPitch()
+        distanceInteractionSwitch.isOn = orbitGeoElementCameraController.isCameraDistanceInteractive
+        // Add an observer to sync the UI for camera heading.
+        headingObservation = orbitGeoElementCameraController.observe(\.cameraHeadingOffset) { [weak self] _, _ in
+            DispatchQueue.main.async {
+                self?.updateUIForHeading()
+            }
+        }
+        isCameraDistanceInteractiveObservation = orbitGeoElementCameraController.observe(\.isCameraDistanceInteractive) { [weak self] cameraController, _ in
+            DispatchQueue.main.async {
+                self?.distanceInteractionSwitch.isOn = cameraController.isCameraDistanceInteractive
+            }
+        }
```

Comment:
A couple things:

- Since the properties being observed are set on the main queue, the change handler will be called on the main thread, so dispatching isn't necessary.
- For the heading observation, `updateUIForHeading()` is being called twice: once for the observation, and then again inside `headingValueChanged(_:)` (on line `76`). I would suggest removing that one and just use the observation to update the UI.

Reply:
The second bug is a nice catch! 🤦

Reply:
The reason I suggested using the observer to update the UI is that you can specify `initial` as an option to update the UI with the initial value. It doesn't really make a difference because we know that no one outside of this view controller modifies `isCameraDistanceInteractive`, but this view controller doesn't know that and shouldn't really know that, so updating the UI to match the current value when it is first displayed is a good practice.

Reply:
Ah now I see what you means! Thanks for the suggestion.